Src/LogData.c

```diff
@@ -36,316 +73,209 @@ void writeToEEPROM(uint8_t* buffer, uint16_t bufferSize, uint16_t memAddress, ui
+        osMutexRelease(data->gpsData_->mutex_);
     }
+    theLogEntry.currentFlightPhase = getCurrentFlightPhase();
+    theLogEntry.tick = HAL_GetTick();
 }
 
-void highFrequencyLogToSdRoutine(AllData* data, char* buffer)
+/**
+ * @brief Used for logging entries to the EEPROM. First updates the struct, then writes it
+ * the wait times are all managed in the main for loop so highFrequencyLog = lowFrequencyLog
+ * @param data, pointer to AllData Struct sent to updateLogEntry to update the logEntry struct
+ */
+void logEntryOnceRoutine(AllData* data)
 {
-    // Get card mounted
-    uint8_t mounted = 0;
-
-    while (!mounted)
-    {
-        // Keep trying, really need to log during this time
-        mounted = (f_mount(&fatfs, "SD:", 1) == FR_OK);
-
-        FlightPhase flightPhase = getCurrentFlightPhase();
-
-        if ( flightPhase != BURN &&
-                flightPhase != COAST &&
-                flightPhase != DROGUE_DESCENT)
-        {
-            // couldn't mount during important phases, too bad :(
-            return;
-        }
-    }
-
-    // Card mounted, start writing at high frequency
-    HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, 1);
-    uint32_t prevWakeTime = osKernelSysTick();
-
-    for (;;)
-    {   // Grab
-        osDelayUntil(&prevWakeTime, FAST_LOG_DATA_PERIOD);
-
-        FlightPhase flightPhase = getCurrentFlightPhase();
-
-        if ( flightPhase != BURN &&
-                flightPhase != COAST &&
-                flightPhase != DROGUE_DESCENT)
-        {
-            // done important phases, unmount card and exit high frequency logging
-            break;
-        }
-
-        buildLogEntry(data, buffer);
-
-        if (f_open(&file, fileName, FA_OPEN_APPEND | FA_READ | FA_WRITE) == FR_OK)
-        {
-            f_puts(buffer, &file);
-            f_close(&file); // close to save the file
-        }
-    }
-
-    HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, 0);
-    f_mount(NULL, "SD:", 1);
+    updateLogEntry(data);
+    writeLogEntryToEEPROM(currMemAddress);
 }
 
 void logDataTask(void const* arg)
 {
     AllData* data = (AllData*) arg;
-    char buffer[500];
-
-    sprintf(
-        buffer,
-        "accelX,"
-        "accelY,"
-        "accelZ,"
-        "gyroX,"
-        "gyroY,"
-        "gyroZ,"
-        "magnetoX,"
-        "magnetoY,"
-        "magnetoZ,"
-        "pressure,"
-        "temperature(100C),"
-        "combustionChamberPressure(1000psi),"
-        "oxidizerTankPressure(1000psi),"
-        "GPS_time,"
-        "GPS_latitude_degrees,"
-        "GPS_latitude_minutes,"
-        "GPS_longitude_degrees,"
-        "GPS_longitude_minutes,"
-        "GPS_altitude,"
-        "currentFlightPhase,"
-        "elapsedTime(ms),"
-        "softwareVersion\n"
-    );
-
+    initializeLogEntry();
+    uint32_t prevWakeTime;
+    clock_t beforeLogTime, afterLogTime, totalTime;
     for (;;)
     {
+        beforeLogTime = clock();
+        logEntryOnceRoutine();
+        afterLogTime = clock();
```

Comment:
Let's use osKernelSysTick() here, it's suppose to return a time in milliseconds as uint_32. This also means that totalTime can be calculated simply as
afterLogTime - beforeLogTime
since they're both in milliseconds already and so is our LOG PERIOD consts.

With this change, we also won't need to include time.h